src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -1459,11 +1467,35 @@ void DatabaseCatalog::dropTableFinally(const TableMarkedAsDropped & table)
         table.table->drop();
     }
 
+    /// Check if we are interested in a particular disk
+    ///   or it is better to bypass it e.g. to avoid interactions with a remote storage
+    auto look_on_disk =
+        [this](DiskPtr disk, std::shared_ptr<MergeTreeData> tbl)
+    {
+        if (!tbl)
+            /// Not a MergeTree - don't skip
+            return true;
+
+        if (tbl->getStoragePolicy()->tryGetVolumeIndexByDiskName(disk->getName()).has_value())
+            /// Disk is actually used - don't skip
+            return true;
+
+        auto is_local_metadata = [&]() { return !(disk->isRemote() && disk->isPlain()); };
+
+        SearchOrphanedPartsDrives mode = (*tbl->getSettings())[MergeTreeSetting::search_orphaned_parts_drives];
+        bool is_look_needed = mode == SearchOrphanedPartsDrives::ANY
+            || (mode == SearchOrphanedPartsDrives::LOCAL && is_local_metadata());
+
+        LOG_TRACE(log, "look_on_disk: mode {}, is_look_needed {}", mode, is_look_needed);
```

Comment:
This logging looks more for developers, consider making it user friendly, i.e.

```suggestion
        LOG_TRACE(log, "Disk {} will be skipped (search_orphaned_parts_drives: {})", ...);
```

Below we have a log message if the disk will be processed, so it is OK to log only skipped disks I think

Actually I think the message should be in the caller not here